### PR TITLE
Fix wasm simulations

### PR DESCRIPTION
**Pull request by alpe**
Fix wasmd simulations so that there are not operations failing during simulations.

Update MsgStoreCode simulation
Update MsgInstantiateContract simulation
Add MsgExecute simulation

Replaces #856

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -116,6 +116,19 @@ jobs:
             cd ./benchmarks
             go test -bench .
 
+  simulations:
+    executor: golang
+    parallelism: 1
+    resource_class: large
```
Comment (pinosu) [inline .circleci/config.yml:122]: 👍

File: app/sim_test.go [hunk 2]
```diff
@@ -190,6 +196,37 @@ func TestAppImportExport(t *testing.T) {
 	// delete persistent tx counter value
 	ctxA.KVStore(app.keys[wasm.StoreKey]).Delete(wasmtypes.TXCounterPrefix)
 
+	// reset contract code index in source DB for comparison with dest DB
+	dropContractHistory := func(s store.KVStore, keys ...[]byte) {
```
Comment (pinosu) [inline app/sim_test.go:200]: nice!

File: x/wasm/simulation/operations.go [hunk 3]
```diff
@@ -130,44 +181,172 @@ func SimulateMsgInstantiateContract(ak types.AccountKeeper, bk simulation.BankKe
...
+type MsgExecuteContractSelector = func(ctx sdk.Context, wasmKeeper WasmKeeper) sdk.AccAddress
+
+// MsgExecutePayloader extension point to modify msg with custom payload
+type MsgExecutePayloader func(msg *types.MsgExecuteContract) error
+
+// MsgExecuteSenderSelector extension point that returns the sender address
+type MsgExecuteSenderSelector func(wasmKeeper WasmKeeper, ctx sdk.Context, contractAddr sdk.AccAddress, accs []simtypes.Account) (simtypes.Account, error)
+
+// SimulateMsgExecuteContract create a execute message a reflect contract instance
+func SimulateMsgExecuteContract(
```
Comment (pinosu) [inline x/wasm/simulation/operations.go:234]: very good work 👍

**Review by pinosu — APPROVED** [review]
lgtm 👍 thanks for fixing all the major issues and implementing MsgExecuteContract simulations . Really good work 💯

File: x/wasm/simulation/genesis.go [hunk 4]
```diff
@@ -8,7 +8,7 @@ import (
 
 // RandomizeGenState generates a random GenesisState for wasm
 func RandomizedGenState(simstate *module.SimulationState) {
-	params := RandomParams(simstate.Rand)
+	params := types.DefaultParams()
```
Comment (Lockwarr) [inline x/wasm/simulation/genesis.go:11]: Why was it decided to use the same default parameters always?

**Pull request merged**